File: Lesson09_OOP/07_Tea/Teabag.java
/*
Copyright Ann Barcomb and Emily Marasco, 2022
Licensed under GPL v3
See LICENSE.txt for more information.
*/

public class Teabag {
    private final String UNKWN = "unknown";
    private final String teaType;
    private final String teaOrigin;
    private final String teaBrand;

    public String getTeaType() { return this.teaType; }
    public String getTeaOrigin() { return this.teaOrigin; }
    public String getTeaBrand() { return this.teaBrand; }

    public Teabag(String teaType) {
        this.teaType = teaType;
        this.teaOrigin = UNKWN;
        this.teaBrand = UNKWN;
    }

    public Teabag(String teaType, String country) {
        this.teaType = teaType;
        this.teaOrigin = country;
        this.teaBrand = UNKWN;
    }

    public Teabag(String teaType, String country, String brand) {
        this.teaType = teaType;
        this.teaOrigin = country;
        this.teaBrand = brand;
    }

    public Teabag() {
        this.teaType = "samdayeon";
        this.teaOrigin = "Korea";
        this.teaBrand = UNKWN;
    }
}

File: Lesson09_OOP/07_Tea/teabag.py
"""
Copyright Khawla Shnaikat and ChatGPT 3.5, 2023
Licensed under GPL v3
See LICENSE.txt for more information.
"""
class Teabag:
    UNKWN = "unknown"

    def __init__(self, tea_type="samdayeon", tea_origin="Korea", tea_brand=UNKWN):
        self.tea_type = tea_type
        self.tea_origin = tea_origin
        self.tea_brand = tea_brand

    def get_tea_type(self):
        return self.tea_type

    def get_tea_origin(self):
        return self.tea_origin

    def get_tea_brand(self):
        return self.tea_brand
